src/config.rs

```diff
@@ -462,24 +471,48 @@ impl TemplateTarget {
     }
 }
 
-fn expand_directories(files: Files) -> Result<Files> {
-    let expanded = files
-        .into_iter()
-        .map(|(from, to)| expand_directory(&from, to).context(format!("expand file {:?}", from)))
+fn expand_directories(config: &Configuration) -> Result<Files> {
+    let expanded = config
+        .files
+        .iter()
+        .map(|(source, target)| {
+            expand_directory(source, target, config).context(format!("expand file {:?}", source))
+        })
         .collect::<Result<Vec<Files>>>()?;
     Ok(expanded.into_iter().flatten().collect::<Files>())
 }
 
 /// If a file is given, it will return a map of one element
 /// Otherwise, returns recursively all the children and their targets
-///  in relation to parent target
-fn expand_directory(source: &Path, target: FileTarget) -> Result<Files> {
-    if fs::metadata(source)
-        .context("read file's metadata")?
-        .is_file()
-    {
+/// in relation to parent target
+fn expand_directory(source: &Path, target: &FileTarget, config: &Configuration) -> Result<Files> {
+    let metadata = fs::metadata(source).with_context(|| {
+        let path_str = source.to_string_lossy();
+        format!("read metadata for '{path_str}'")
+    })?;
+
+    // Per the File docs, the most reliable way to determine if something
+    // is a file is to simply try to open it. This allows linking to
+    // other symlinks and other readable special files.
+    let is_readable_file = !metadata.is_dir() && fs::File::open(source).is_ok();
```

Comment:
`metadata.is_dir()` should traverse symlinks and query the target of the chain. What edge case is this catching?

Reply:
I think I was just being extra cautious, trying not to assume that if a file was not a directory that it must be a readable file. But thinking through it some more, I guess that's kind of irrelevant with regard to whether to make a symlink. If the symlink points to a file that's not readable for whatever reason, that's probably not something dotter needs to concern itself with. I'll change this just to be based on whether it's a directory.